### PR TITLE
feature: DisableOverlay component

### PR DESCRIPTION
**Pull request by moiskillnadne**

### CODE REVIEW

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: src/entities/activity/lib/hooks/useActivitiesByIds.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+export const useActivitiesByIds = ({ appletDetails }: Props) => {
+  const activitiesIds = useMemo(() => {
+    return appletDetails.activities.map(x => x.id)
+  }, [appletDetails])
+
+  const queryResults = useActivitiesByIdsQuery({ activitiesIds })
+
+  const { isError, isLoading, data } = useMemo(() => {
+    const isLoading = queryResults.reduce((acc, item) => {
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/entities/activity/lib/hooks/useActivitiesByIds.ts:19]: Done, thank you!

File: src/entities/activity/lib/hooks/useActivitiesByIds.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      return item.isLoading || acc
+    }, false)
+
+    const isError = queryResults.reduce((acc, item) => {
+      return item.isError || acc
+    }, false)
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/entities/activity/lib/hooks/useActivitiesByIds.ts:25]: Done

File: src/entities/activity/lib/hooks/useSupportableActivities.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+type Props = {
+  appletDetails: AppletDetailsDTO
+}
+
+export const useSupportableActivities = ({ appletDetails }: Props) => {
+  const { data, isError, isLoading } = useActivitiesByIds({ appletDetails })
+
+  const activities = useMemo(() => {
+    return data.filter(x => x) as Array<ActivityDTO>
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/entities/activity/lib/hooks/useSupportableActivities.ts:16]: Data is Array<ActivityDTO | undefined> type but I need get only Array<ActivityDTO>

File: src/entities/activity/model/activityBuilder.ts [hunk 4]
```diff
@@ -96,6 +97,23 @@ class ActivityBuilder {
     }
   }
 
+  public checkIsSupportableActivity(activities: ActivityDTO[]): SupportableActivities | null {
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/entities/activity/model/activityBuilder.ts:100]: Done

File: src/entities/activity/model/activityBuilder.ts [hunk 5]
```diff
@@ -96,6 +97,23 @@ class ActivityBuilder {
...
+        const isItemSupported = supportableItemTypes.includes(itemType)
+
+        return acc && isItemSupported
+      }, true)
+
+      acc[activityId] = isSupportable
+
+      return acc
+    }, {})
+  }
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/entities/activity/model/activityBuilder.ts:115]: I think your suggested way will add extra properties and dependencies between modules/hooks.
Extra dependencies make code difficult to read and support

**Review by BamMironov — APPROVED** [review]

**Pull request merged**